nighres/statistics/segmentation_statistics.py

```diff
@@ -0,0 +1,180 @@
+
+        if overwrite is True:
+            # delete current stats file to start from the beginning
+            os.remove(csv_file)
+    else:
+        csv_file = output_csv
+
+    # start virtual machine, if not already running
+    try:
+        mem = _check_available_memory()
+        nighresjava.initVM(initialheap=mem['init'], maxheap=mem['max'])
+    except ValueError:
+        pass
+    # create algorithm instance
+    stats = nighresjava.StatisticsSegmentation()
+
+    # load first image and use it to set dimensions and resolution
+    img = load_volume(segmentation)
+    data = img.get_data()
+    affine = img.get_affine()
+    header = img.get_header()
+    resolution = [x.item() for x in header.get_zooms()]
+    dimensions = data.shape
+
+    stats.setDimensions(dimensions[0], dimensions[1], dimensions[2])
+    stats.setResolutions(resolution[0], resolution[1], resolution[2])
+
+    stats.setSegmentationImage(nighresjava.JArray('int')(
+                                    (data.flatten('F')).astype(int).tolist()))
+    stats.setSegmentationName(_fname_4saving(rootfile=segmentation))
+
+    # other input images, if any
+    if intensity is not None:
+        data = load_volume(intensity).get_data()
+        stats.setIntensityImage(nighresjava.JArray('float')(
+                                    (data.flatten('F')).astype(float)))
+        stats.setIntensityName(_fname_4saving(rootfile=intensity))
+    
+    if template is not None:
+        data = load_volume(template).get_data()
+        stats.setTemplateImage(nighresjava.JArray('int')(
+                                    (data.flatten('F')).astype(int).tolist()))
+        stats.setTemplateName(_fname_4saving(rootfile=template))
+    
+    # set algorithm parameters
+    if atlas is not None:
+        stats.setAtlasFile(atlas)
+        
+    stats.setSkipFirstLabel(skip_first)
+    stats.setIgnoreZeroIntensities(ignore_zero)
+    
+    if len(statistics)>0: stats.setStatistic1(statistics[0])
+    if len(statistics)>1: stats.setStatistic2(statistics[1])
+    if len(statistics)>2: stats.setStatistic3(statistics[2])
+
+    stats.setSpreadsheetFile(csv_file)
+    
+    # execute the algorithm
+    try:
+        stats.execute()
+
+    except:
+        # if the Java module fails, reraise the error it throws
+        print("\n The underlying Java code did not execute cleanly: ")
+        print(sys.exc_info()[0])
+        raise
+        return
+
+    # reshape output to what nibabel likes
+    output = False
+    for st in statistics: 
+        if st=="Cluster_maps": 
+            output=True
+    
+    if (output):
+        data = np.reshape(np.array(stats.getOutputImage(),
+                                       dtype=np.int32), dimensions, 'F')
+        header['cal_min'] = np.nanmin(data)
+        header['cal_max'] = np.nanmax(data)
+        output = nb.Nifti1Image(data, affine, header)
+
+        if save_data:
+            save_volume(map_file, output)
```

Comment:
Is the csv file also saved / can the path where it is saved be chosen (like for the map)? what is the name?

Reply:
the csv file is saved internally, in the Java part. I'll check that I pass it around too

Reply:
Everything is correct in there :)